### PR TITLE
🔨  Refactor Cask exceptions.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Also fixes https://github.com/caskroom/homebrew-cask/issues/25095.

### CODE REVIEW

File: Library/Homebrew/cask/lib/hbc/exceptions.rb [hunk 1]
```diff
@@ -3,56 +3,40 @@ class CaskError < RuntimeError; end
...
     def to_s
-      "#{token} is not installed"
+      "Cask '#{token}' is not installed."
     end
   end
 
   class CaskUnavailableError < AbstractCaskErrorWithToken
     def to_s
-      "No available Cask for #{token}"
+      "Cask '#{token}' is unavailable" << (reason.empty? ? "." : ": #{reason}")
```
Comment (apjanke) [inline Library/Homebrew/cask/lib/hbc/exceptions.rb]: Nitpick: should this be `": #{reason}."` for consistency, so all error messages end with a "."?

File: Library/Homebrew/cask/lib/hbc/exceptions.rb [hunk 2]
```diff
@@ -101,60 +85,67 @@ def to_s
...
   class CaskInvalidError < AbstractCaskErrorWithToken
-    attr_reader :submsg
-    def initialize(token, *submsg)
-      super(token)
-      @submsg = submsg.join(" ")
-    end
-
     def to_s
-      "Cask '#{token}' definition is invalid#{": #{submsg}" unless submsg.empty?}"
+      "Cask '#{token}' definition is invalid" << (reason.empty? ? ".": ": #{reason}")
```
Comment (apjanke) [inline Library/Homebrew/cask/lib/hbc/exceptions.rb]: Should this also be `: #{reason}."`?

File: Library/Homebrew/cask/lib/hbc/exceptions.rb (hunk 1 above)
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/exceptions.rb]: The `reason` should include the `.` itself.

**Comment by reitermarkus** [comment]
Altered some more messages.

> I'm also getting this error.

`MacOS.release` was renamed to `MacOS.version` some time ago, seems like you still have some old Casks installed which are using `MacOS.release`.

File: Library/Homebrew/cask/lib/hbc/exceptions.rb (hunk 1 above)
Comment (apjanke) [inline Library/Homebrew/cask/lib/hbc/exceptions.rb]: Ah, got it. Oversight on my part.

**Comment by apjanke** [comment]
> MacOS.release was renamed to MacOS.version some time ago, seems like you still have some old Casks installed which are using MacOS.release.

That was basically it. My `Library/Taps/caskroom/homebrew-cask` was sitting on an old branch instead of `master`. Thanks.

**Comment by apjanke** [comment]
Okay. I'm seeing reasons in the error messages as expected now.

```
$ mkdir /tmp/foo.rb
[/usr/local/Homebrew/Library/Taps on ⇄ pr-2771]
$ brew cask install /tmp/foo.rb
Error: Cask 'foo' is unavailable: '/tmp/foo.rb' is not a file.
Error: nothing to install
[✘ /usr/local/Homebrew/Library/Taps on ⇄ pr-2771]
$ brew cask install bogus-cask
Error: Cask 'bogus-cask' is unavailable: No Cask with this name exists.
Error: nothing to install
[✘ /usr/local/Homebrew/Library/Taps on ⇄ pr-2771]
$ brew cask install bogus-cask /tmp/foo.rb http://www.example.com/bogus/nonexistent-cask.rb
Error: Cask 'bogus-cask' is unavailable: No Cask with this name exists.
Error: Cask 'foo' is unavailable: '/tmp/foo.rb' is not a file.
==> Downloading http://www.example.com/bogus/nonexistent-cask.rb.

curl: (22) The requested URL returned error: 404 Not Found
Error: Cask 'nonexistent-cask' is unavailable: Failed to download http://www.example.com/bogus/nonexistent-cask.rb.
Error: nothing to install
```

**Comment by apjanke** [comment]
How about having `info` and `uninstall` (and maybe `audit` and `fetch` and `zap`) give warnings with suggestions the way `install` and `reinstall` do?

```
[/usr/local/Homebrew/Library/Homebrew/cask on ⇄ pr-2771]
$ brew cask install powermat
Error: Cask 'powermat' is unavailable: No Cask with this name exists. Did you mean one of:
powermate
Error: nothing to install
[✘ /usr/local/Homebrew/Library/Homebrew/cask on ⇄ pr-2771]
$ brew cask info powermat
Error: Cask 'powermat' is unavailable: No Cask with this name exists.
[✘ /usr/local/Homebrew/Library/Homebrew/cask on ⇄ pr-2771]
$ brew cask uninstall powermat
Error: Cask 'powermat' is unavailable: No Cask with this name exists.
[✘ /usr/local/Homebrew/Library/Homebrew/cask on ⇄ pr-2771]
$ brew cask reinstall powermat
Error: Cask 'powermat' is unavailable: No Cask with this name exists. Did you mean one of:
powermate
Error: nothing to install
[✘ /usr/local/Homebrew/Library/Homebrew/cask on ⇄ pr-2771]
$
```

That always happens around a `cask = CaskLoader.load(cask_token)` call with a rescue around it. Maybe `Hbc::CLI` could get a `load_cask_with_suggestions` method to wrap it for convenience?

**Comment by reitermarkus** [comment]
@apjanke, added a `casks` method to iterate over Casks given as arguments.

**Pull request merged**